homeassistant/components/mysensors/gateway.py

```diff
@@ -58,48 +60,103 @@ def is_socket_address(value):
         raise vol.Invalid("Device is not a valid domain name or ip address") from err
 
 
-def get_mysensors_gateway(hass, gateway_id):
-    """Return MySensors gateway."""
-    if MYSENSORS_GATEWAYS not in hass.data:
-        hass.data[MYSENSORS_GATEWAYS] = {}
-    gateways = hass.data.get(MYSENSORS_GATEWAYS)
+async def try_connect(hass: HomeAssistantType, user_input: Dict[str, str]) -> bool:
+    """Try to connect to a gateway and report if it worked."""
+    if user_input[CONF_DEVICE] == MQTT_COMPONENT:
+        return True  # dont validate mqtt. mqtt gateways dont send ready messages :(
+    user_input_copy = user_input.copy()
+    try:
+        gateway: Optional[BaseAsyncGateway] = await _get_gateway(
+            hass, user_input_copy, str(randint(0, 10 ** 6)), persistence=False
```

Comment:
We shouldn't generate an id with randint.

Reply:
This id is very short lived (only in `try_connect()`) and does not get stored anywhere.

What should i use instead?

Reply:
The gateway doesn't require a random string to be created.